synapse/federation/transaction_queue.py

```diff
@@ -303,9 +303,13 @@ def _attempt_new_transaction(self, destination):
             )
             return
 
+        pending_pdus = []
         try:
             self.pending_transactions[destination] = 1
 
+            if not (yield would_retry(destination, self.clock, self.store)):
```

Comment:
given we logger.debug when we get a `NotRetryingDestination` exception (a block of code which is now almost redundant, btw), should we do so here? should we in fact throw a NotRetryingDestination here?